scripts/ci/coverage/coverage_analysis.py

```diff
@@ -0,0 +1,505 @@
+                                        test_suite = {
+                                            "name":testsuite['name'],
+                                            "path":testsuite['path'],
+                                            "platforms":[],
+                                            "runnable": testsuite['runnable'],
+                                            "status":"",
+                                            "test_cases":[]
+                                        }
+                                        test_case  = {
+                                            "name": testcase_name
+                                        }
+                                        if 'qemu' in testsuite['platform'] or 'native' in testsuite['platform']:
+                                            if test_suite['status'] == "":
+                                                test_suite['status'] = 'sim_only'
+
+                                            if test_suite['status'] == 'hw_only':
+                                                test_suite['status'] = 'mixed'
+                                        else:
+                                            if test_suite['status'] == "":
+                                                test_suite['status'] = 'hw_only'
+
+                                            if test_suite['status'] == 'sim_only':
+                                                test_suite['status'] = 'mixed'
+                                        test_suite['test_cases'].append(test_case)
+                                        test_suite['platforms'].append(testsuite['platform'])
+                                        sub_component["test_suites"].append(test_suite)
+                                    else:
+                                        test_case = {}
+                                        test_cases = test_suite['test_cases']
+                                        known_testcase_flag = False
+                                        for i_testcase in test_cases:
+                                            if testcase_name == i_testcase['name']:
+                                                test_case = i_testcase
+                                                known_testcase_flag = True
+                                                break
+                                        if known_testcase_flag is False:
+                                            test_case = {
+                                                "name":testcase_name
+                                            }
+                                            test_suite['test_cases'].append(test_case)
+            file.close()
+
+    def get_files_from_maintainers_file(self, component_name):
+        files_path = []
+        for item in self.maintainers_file:
+            _found_flag = False
+            try:
+                tests = self.maintainers_file[item]['tests']
```

Comment:
that will fail if an area does not have tests....
You should look into using the maintainer file parser, see `scripts/get_maintainer.py` then 
`from get_maintainer import Maintainers`